### PR TITLE
Clean ThreadFragment `observers` code

### PR DESCRIPTION
**Pull request by KevinBoulongne**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 1]
```diff
@@ -159,30 +160,31 @@ class ThreadFragment : Fragment() {
...
+        Log.i("UI", "Received messages (${messages.size})")
+        if (messages.isEmpty()) leaveThread()
+    }
+
+    private fun leaveThread() {
+        threadViewModel.deleteThread(navigationArgs.threadUid)
+        // TODO: The day we'll have the Notifications, we'll have to check if this `popBackStack` executes correctly.
+        // TODO: If the fact of opening a Thread via a Notification doesn't fully populate the backStack, the action
+        // TODO: of leaving this fragment (either via a classic Back button, or via this `popBackStack`) will
+        // TODO: probably quit the app instead of going back to the ThreadList fragment (as it should be).
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:186]: It won't quit the app, it simply won't do anything instead

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt]: Comment has been updated.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 2]
```diff
@@ -158,30 +159,31 @@ class ThreadFragment : Fragment() {
...
+        if (messages.isEmpty()) leaveThread()
+    }
+
+    private fun leaveThread() {
+        threadViewModel.deleteThread(navigationArgs.threadUid)
+        // TODO: The day we'll have the Notifications, this `popBackStack` will probably fail to execute correctly.
+        // TODO: When opening a Thread via a Notification, the action of leaving this fragment
+        // TODO: (either via a classic Back button, or via this `popBackStack`) will probably
+        // TODO: do nothing instead of going back to the ThreadList fragment (as it should be).
+        findNavController().popBackStack(R.id.threadListFragment, inclusive = false)
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:186]: To verify if it's always necessary

File: build.gradle [hunk 3]
```diff
@@ -1,4 +1,5 @@
 buildscript {
+    ext.java_version = JavaVersion.VERSION_1_8
```
Comment (sirambd) [inline build.gradle:2]:
```suggestion
    ext.java_version = JavaVersion.VERSION_11
```

**Review by sirambd — CHANGES_REQUESTED** [review]
Waiting for request changes

**Pull request merged**